### PR TITLE
Initial Untracked Transactions Viewing & Updating

### LINKED ISSUE #107: [Transactions Page] Displaying Untracked Transactions

### PR DESCRIPTION
**Pull request by johnsonav1992**
This is the first iteration of this. We can redesign as needed.

### CODE REVIEW

File: src/app/services/transaction.service.ts [hunk 1]
```diff
@@ -10,29 +9,39 @@ import {
...
 export class TransactionService {
+    baseUrl = `${BE_API_URL}/transactions`;
+
     http = inject(HttpClient);
     authService = inject(AuthService);
-    baseUrl = `${BE_API_URL}/transactions`;
 
     currentSelectedLineItem = signal<SelectedLineItem | null>(null);
-    newlyCreatedTransactionId = new Subject<string>();
+    newlyCreatedTransactionId = signal<string | null>(null);
```
Comment (eliotmjohnson) [inline src/app/services/transaction.service.ts]: Could you revert this back to a Subject? I had a specific implementation with it and I don't want to use signals here.

File: src/app/services/line-item.service.ts [hunk 2]
```diff
@@ -70,7 +69,7 @@ export class LineItemService {
     fetchLineItem(lineItemId: string): LineItem | undefined {
         if (
             !this.fetchedLineItem ||
```
Comment (eliotmjohnson) [inline src/app/services/line-item.service.ts]: This should be invoked as well

File: src/app/components/transaction-modal/transaction-modal.component.ts [hunk 3]
```diff
@@ -306,41 +313,42 @@ export class TransactionModalComponent implements AfterViewInit, OnInit {
...
     }
 
     updateEagerTransactionId(newTransaction: NewTransaction) {
-        this.transactionService.newlyCreatedTransactionId.subscribe((id) => {
-            const newEagerTransaction = this.lineItemService
+        const newlyCreatedTransactionId =
+            this.transactionService.newlyCreatedTransactionId();
+
+        const newEagerTransaction =
+            newTransaction.lineItemId &&
```
Comment (eliotmjohnson) [inline src/app/components/transaction-modal/transaction-modal.component.ts]: Could you revert this back to a Subject? I had a specific implementation with it and I don't want to use signals here.

File: src/app/services/transaction.service.ts [hunk 4]
```diff
@@ -71,7 +86,7 @@ export class TransactionService {
                             new Date(transaction.date)
                         );
                     } else {
-                        this.newlyCreatedTransactionId.next(
+                        this.newlyCreatedTransactionId.set(
```
Comment (eliotmjohnson) [inline src/app/services/transaction.service.ts]: Here as well

**Review by eliotmjohnson — CHANGES_REQUESTED** [review]

File: src/app/components/transaction-modal/transaction-modal.component.ts (hunk 3 above)
Comment (johnsonav1992) [reply src/app/components/transaction-modal/transaction-modal.component.ts]: Oh sorry! Haha I did this to test something and then forgot about it. Whoops!

**Review by eliotmjohnson — APPROVED** [review]

**Pull request merged**